src/include/multipath/complex.rb

```diff
@@ -49,7 +50,9 @@ def initialize_multipath_complex(include_target)
       @service_status = 0
       @has_dumbtab = false
       @device_template = "vendor %1; product %2"
-      @builtin_multipath_conf_path = "/tmp/.yast2-multipath-builtin-conf"
+
+      tfile = Tempfile.new('.yast2-multipath-builtin-conf')
```

Comment:
The [Tempfile documentations](https://ruby-doc.org/stdlib-2.5.1/libdoc/tempfile/rdoc/Tempfile.html) says:

> When a Tempfile object is garbage collected, or when the Ruby interpreter exits, its associated temporary file is automatically deleted.

Because you assign the `Tempfile` object to a local variable it can be garbage collected anytime after leaving this function. Which means an attacker would have a short period of time to create the same file before YaST really uses  it later.

To avoid this keep the `Tempfile` object until it's not needed anymore, e.g.: 
```suggestion
      @tfile = Tempfile.new('.yast2-multipath-builtin-conf')
```

Although the file will be removed automatically at exit it would be nice to remove it when it's not needed anymore with

```ruby
@tfile.close
@tfile.unlink
```

I do not know when it's not needed, check the code, it seems after calling the `Read_MultipathConfig` is should be safe to remove it. But if it might be reused again later or there is some problem with the removal you can leave it for the automatic cleanup at exit. I assume the saved file is not big so it won't block much disk space.

Reply:
Thanks for the suggestion of changing of tfile, fix in the next commit.

```
@tfile.close
@tfile.unlink
```
The legacy code of ``RegisterAgent(..., @builtin_multipath_conf_path, ...)`` is not pretty at all,It is used in Read_Configures() for 5 Read_* functions, can only be removed before Read_Configures() exit. 
Since @tfile is not defined in this function, close here is make sense but slightly abrupt like you said. I prefer to leave it with automatically cleanup. And yes, the file is not big, around 20k.